src/events/EnvelopingEventEmitter.ts

```diff
@@ -2,35 +2,32 @@ import EventEmitter from 'events';
 
 const events = {
   init: 'init',
-  enveloping: 'enveloping',
   'refresh-relays': 'refresh-relays', // TODO validate if its needed, since we do not refresh relays anymore
   'refreshed-relays': 'refreshed-relays', // TODO validate if its needed, since we do not refresh relays anymore
   'next-relay': 'next-relay',
   'sign-request': 'sign-request',
   'validate-request': 'validate-request',
   'send-to-relayer': 'send-to-relayer',
-  'relayer-response': 'relayer-response', // FIXME needs to have a boolean flag
+  'relayer-response': 'relayer-response',
 } as const;
 
 type Event = keyof typeof events;
 
 class EnvelopingEventEmitter extends EventEmitter {
   registerEventListener(
-    eventType: Event,
-    eventHandle: (...args: unknown[]) => void
+    eventHandler: (event: Event, ...args: unknown[]) => void
   ) {
-    this.on(eventType, eventHandle);
+    this.on('enveloping', eventHandler);
```

Comment:
Why are we hard-coding this event here?

Reply:
By using the event type, you will be just listening to that event when you register.  The current implementation allows you to listen to all of the envelopingEvents together, while in the previous implementation you were only listening to each event, I modified to keep the same behavior as before.

Reply:
the enveloping its no an event, its like the wrapper of the events.

Reply:
updated

Reply:
> the enveloping its no an event, its like the wrapper of the events.

Yes, I was missing this 👍🏼